plugins-mu/hm-stack/class-endpoint-controller.php

```diff
@@ -0,0 +1,238 @@
+<?php
+/**
+ * Build endpoint data for HM Stack data on this site.
+ *
+ * @package HMStackIntegration
+ */
+
+namespace HM_Stack;
+
+use WP_REST_Response;
+use WP_Error;
+use WP_Http;
+
+/**
+ * Class Endpoint_Controller
+ */
+class Endpoint_Controller extends \WP_REST_Controller {
+	/**
+	 * Register HM_Stack data routes for the WP REST API.
+	 *
+	 * @todo:: Add schemas to each endpoint.
+	 */
+	public function register_routes() {
+		$namespace = 'hm-stack/v1';
+
+		// Fetch all activity/alerts against this site.
+		register_rest_route( $namespace, 'activity', [
+			'methods'  => \WP_REST_Server::READABLE,
+			'callback' => [ $this, 'get_site_activity' ],
+			'args'     => [
+				'context' => [
+					'default' => 'view',
+				],
+			],
+			'permission_callback' => [ $this, 'permissions_check' ],
+		] );
+
+		// Fetch bandwidth usage for this site.
+		register_rest_route( $namespace, 'bandwidth-usage', [
+			'methods'  => \WP_REST_Server::READABLE,
+			'callback' => [ $this, 'get_bandwidth' ],
+			'args'     => [
+				'context' => [
+					'default' => 'view',
+				],
+			],
+			'permission_callback' => [ $this, 'permissions_check' ],
+		] );
+
+		// Fetch all site environmental data for this site.
+		register_rest_route( $namespace, 'environment-data', [
+			'methods'  => \WP_REST_Server::READABLE,
+			'callback' => [ $this, 'get_environment' ],
+			'args'     => [
+				'context' => [
+					'default' => 'view',
+				],
+			],
+			'permission_callback' => [ $this, 'permissions_check' ],
+			'schema'              => [
```

Comment:
The typical pattern is to have a `get_item_schema` but it's probab;y fine either way